recipes/mimalloc/all/conanfile.py

```diff
@@ -0,0 +1,201 @@
+            cmake.build()
+
+    def package(self):
+        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
+        with tools.vcvars(self.settings) if self.settings.compiler == "Visual Studio" else tools.no_op():
+            cmake = self._configure_cmake()
+            cmake.install()
+
+        if self.options.get_safe("single_object"):
+            tools.remove_files_by_mask(os.path.join(self.package_folder, "lib"),
+                                       "*.a")
+            shutil.copy(os.path.join(self.package_folder, "lib", self._obj_name + ".o"),
+                        os.path.join(self.package_folder, "lib", self._obj_name))
+
+        if self.settings.os == "Windows" and self.options.shared:
+            if self.settings.arch == "x86_64":
+                self.copy("mimalloc-redirect.dll", src=os.path.join(self._source_subfolder, "bin"),
+                          dst="bin")
+            elif self.settings.arch == "x86":
+                self.copy("mimalloc-redirect32.dll", src=os.path.join(self._source_subfolder, "bin"),
+                          dst="bin")
+
+    @property
+    def _obj_name(self):
+        name = "mimalloc"
+        if self.options.secure:
+            name += "-secure"
+        if self.settings.build_type not in ("Release", "RelWithDebInfo", "MinSizeRel"):
+            name += "-{}".format(str(self.settings.build_type).lower())
+        return name
+
+    @property
+    def _lib_name(self):
+        name = "mimalloc" if self.settings.os == "Windows" else "libmimalloc"
+
+        if self.settings.os == "Windows" and not self.options.shared:
+            name += "-static"
+        if self.options.secure:
+            name += "-secure"
+        if self.settings.build_type not in ("Release", "RelWithDebInfo", "MinSizeRel"):
+            name += "-{}".format(str(self.settings.build_type).lower())
+        return name
+
+    def package_info(self):
+        if self.options.get_safe("inject"):
+            self.cpp_info.includedirs = []
+            self.cpp_info.libdirs = []
+            self.cpp_info.resdirs = []
+            return
+
+        if self.options.get_safe("single_object"):
+            obj_ext = "o"
+            obj_file = "{}.{}".format(self._obj_name, obj_ext)
+            obj_path = os.path.join(self.package_folder, "lib", obj_file)
+            self.cpp_info.exelinkflags = [obj_path]
+            self.cpp_info.sharedlinkflags = [obj_path]
+            self.cpp_info.libdirs = []
+            self.cpp_info.bindirs = []
+        else:
+            self.cpp_info.libs = tools.collect_libs(self)
+
+        if self.settings.os == "Linux":
+            self.cpp_info.system_libs.append("pthread")
+        if not self.options.shared:
+            if self.settings.os == "Windows":
+                self.cpp_info.system_libs.extend(["psapi", "shell32", "user32", "bcrypt"])
+            elif self.settings.os == "Linux":
+                self.cpp_info.system_libs.append("rt")
```

Comment:
In the case of `inject`, you will not run this code.
Is it alright?

Reply:
Yes. The `inject` option models the situation in which the user doen't want to change their software at all... In this mode of operation the user must load mimalloc with `LD_PRELOAD`, for instance, if it's on Linux.

Reply:
Great, thanks for the good explanation.